Tests/Basic/OutputByteStreamTests.swift

```diff
@@ -25,30 +25,69 @@ class OutputByteStreamTests: XCTestCase {
         let streamable: Streamable = Character("!")
         stream.write(streamable)
 
-        stream.flush()
         
         XCTAssertEqual(stream.position, "Hello, world!".utf8.count)
+        stream.flush()
         XCTAssertEqual(stream.bytes, "Hello, world!")
     }
     
     func testStreamOperator() {
-        let stream = OutputByteStream()
+        let stream = BufferedOutputByteStream()
 
         let streamable: Streamable = Character("!")
         stream <<< "Hello" <<< Character(",") <<< Character(" ") <<< [UInt8]("wor".utf8) <<< [UInt8]("world".utf8)[3..<5] <<< streamable
         
         XCTAssertEqual(stream.position, "Hello, world!".utf8.count)
         XCTAssertEqual(stream.bytes, "Hello, world!")
 
-        let stream2 = OutputByteStream()
+        let stream2 = BufferedOutputByteStream()
         stream2 <<< (0..<5)
         XCTAssertEqual(stream2.bytes, [0, 1, 2, 3, 4])
     }
     
+    func testBufferCorrectness() {
+        let bytes1k = [UInt8](repeating: 0, count: 1 << 10)
+        let bytes4k = [UInt8](repeating: 0, count: 1 << 12)
+        let bytes8k = [UInt8](repeating: 0, count: 1 << 13)
+        let bytes16k = [UInt8](repeating: 0, count: 1 << 14)
```

Comment:
We should make these have some kind of contents, otherwise this test isn't really checking anything other than the result counts (and for complete garbage copies).